### DISCUSSION TITLE
403 error on files in public/storage

### DISCUSSION THREAD

**Discussion by tylerpixelguild** [laravel/framework#34675] [Q&A]
All of the files uploaded to public/storage return a 403 error when requested in a browser. This includes the profile photo I uploaded in the jetstream profile page. I've verified the files are on the server and they exist and are able to be opened and viewed. I don't get the error on my local copy but it happens on production.

- I have the appropriate URL in the APP_URL variable in the .env file. 
- I ran the php artisan storage:link and it executed without error and works.  
- The directory and files are owned by the web server group and are the same as the "js" and "css" directories in public which are accessible through a browser.

This is the "disks" section of my filesystems.php config file.

```
    'disks' => [

        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
        ],

        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
        ],

        's3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
        ],

    ],
```

Any ideas of what the issue could be?

**Comment by patake** [comment]
I got exactly the same problem here.

Did you solve the problem? Can you share the way to fix it?

Thanks in advance

**Comment by v0idwalker** [comment]
While it might be a different issue altogether, my issue while struggling with the 403 on storage was twofold:
1. my hosting provider had different mounting points for the apache service and for the user who can handle the hosting, which caused that the symlink generated by ```php artisan storage:link``` to point to a different (nonexistent) folder from the apache's perspective. I removed the link and manually created a relative link from the public folder ```ln -s ../storage/app/public storage```
2. I had a (user-me caused) problem where the folder public contained a folder "storage" which should not be a folder but a symbolic link generated by aforementioned artisan command. This would admittedly and most probably not result in 403, but an empty folder as all the data would be in storage/app/public with no connection to public/storage folder.
